.github/workflows/merge-group.yml

```diff
@@ -63,3 +65,40 @@ jobs:
       - name: Linux Gadget Tests w/o debug assertions
         run: |
           cargo nextest run --profile ci --workspace --cargo-profile dev-no-assertions -E 'test(circuit::gadgets)'
+
+  # TODO: Make this a required status check
+  gpu-benchmark:
+    name: Run fibonacci bench on GPU
+    runs-on: gpu-bench-a6000
+    env:
+      NVIDIA_VISIBLE_DEVICES: all
+      NVIDIA_DRIVER_CAPABILITITES: compute,utility
+      EC_GPU_FRAMEWORK: cuda
+    steps:
+      # Checkout and install deps
+      - uses: actions/checkout@v4
+      - uses: actions-rs/toolchain@v1
+      - uses: Swatinem/rust-cache@v2
+      - uses: taiki-e/install-action@just
+
+      # Set up GPU
+      # Check we have access to the machine's Nvidia drivers
+      - run: nvidia-smi
+      # The `compute`/`sm` number corresponds to the Nvidia GPU architecture
+      # In this case, the self-hosted machine uses the Ampere architecture, but we want this to be configurable
+      # See https://arnon.dk/matching-sm-architectures-arch-and-gencode-for-various-nvidia-cards/
+      - name: Set env for CUDA compute
+        run: echo "CUDA_ARCH=$(nvidia-smi --query-gpu=compute_cap --format=csv,noheader | sed 's/\.//g')" >> $GITHUB_ENV
+      - name: set env for EC_GPU
+        run: echo 'EC_GPU_CUDA_NVCC_ARGS=--fatbin --gpu-architecture=sm_${{ env.CUDA_ARCH }} --generate-code=arch=compute_${{ env.CUDA_ARCH }},code=sm_${{ env.CUDA_ARCH }}' >> $GITHUB_ENV
+      - run: echo "${{ env.EC_GPU_CUDA_NVCC_ARGS}}"
```

Comment:
If we're going to probe for these, we should make the connection between these and `just`, either through the line that gets executed or the env file it's executed with. 

*One* way to perform this is to use the `bench.env` file set up in this PR, and append those variables to the `.env`. Since this bench run is ephemeral, `bench.env` contains default bench options, to which this probing appends the cuda compilation options.